syntax = "proto3";

import "generic_message.proto"; // Empty
import "heartbeat.proto"; // HeartbeatReply
import "command.proto"; // CommandMessage

// Sequencer Service Definition
service Sequencer {
  // Enqueue a sequence command of varying priority
  rpc EnqueueCommand (CommandMessage) returns (HeartbeatReply) {}

  // Heart beat message for clients
  rpc RequestHeartBeat(Empty) returns (HeartbeatReply) {}
}
